lib/react_client/react_interop.dart

```diff
@@ -34,8 +36,54 @@ abstract class React {
 
   external static bool isValidElement(dynamic object);
   external static ReactClass get Fragment;
+
+  /// Returns a ref that can be attached to an element via the ref attribute.
+  ///
+  /// When a ref is passed to an element in render, a reference to the node becomes
+  /// accessible at the 'current' attribute of the ref.
+  ///
+  /// See: <https://reactjs.org/docs/refs-and-the-dom.html#creating-refs>.
+  external static createRef();
 }
 
+class createRef<CurrentType> {
+  var jsCreateRef;
+
+  createRef() {
+    jsCreateRef = React.createRef();
+  }
+
+  CurrentType get current {
+    var jsCurrent = getProperty(jsCreateRef, 'current');
```

Comment:
This may be worth another try but I think I mentioned this today, this approach was actually the very first thing we tried but because js side react does that object.freeze it actually has panic attacks if you mess around with that at all and I’m not sure what or how but I believe the interceptors are part of it and change the object and so then react looses its mind if when it sees not it’s own object value.